### PR TITLE
Myshrimp

### PR DESCRIPTION
**Pull request by Myshrimp**

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4941) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4941) before we can accept your contribution.<br/><hr/>**Liu Jiawei** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=4941) it.</sub>

File: src/simplemath.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#include "platform.h"
+#ifdef NCNN_SIMPLEMATH
+#include <stdlib.h>
+#include<stdio.h>
+float absolute(float x)//����ֵ
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:5]: The encoding can be improved. UTF-8 encoding recommended. Spaces around `//` prefered.

i.e.
```cpp
float absolute(float x) // your_comment
```

File: src/simplemath.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+{
+	if (x < 0)x = -x;
+	return x;
+}
+
+float Factorial(int x)//�׳�
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:11]: ditto

File: src/simplemath.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+{
+	if (x == 1 || x == 0)return 1;
+	else
+		return 1.0 * x * Factorial(x - 1);
+}
+
+
+float nth(float x, int n)//n�η�
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:19]: ditto

File: src/simplemath.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	if (n == 0)
+	{
+		return 1;
+	}
+	if (n < 0)
+	{
+		return (1 / x) * nth(x, n + 1);
+	}
+}
+float Bernoulli(int x)//��Ŭ����
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:34]: ditto

File: src/simplemath.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		{
+			while (k)
+			{
+				k--;
+				B += -1.0 * (Factorial(x) * Bernoulli(k)) / (Factorial(x - k) * Factorial(k) * (x - k + 1));
+			}
+			return B;
+		}
+}
+float tan(float x)//tan(x)  ���ȱ���0.000001 �����ٶ�̫��
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:55]: ditto

File: src/simplemath.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	int i = 1;
+	float e = 1, sum = 0;
+	while (x < (-pai / 2))x += pai;
+	while (x > (pai / 2))x -= pai;
+	if (x == (pai / 2))
+	{
+		printf("\tNaN\n");
+		return 0;
+	}
+	while (absolute(e) > accuracy && i <= 24)//���Ǽ����ٶȣ�����Bernoulli()
```
Comment (zchrissirhcz) [inline src/simplemath.cpp:66]: ditto

File: src/simplemath.h [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#ifndef TAN_H
+#define TAN_H
+#include<stdio.h>
+#define pai 3.14159265
```
Comment (zchrissirhcz) [inline src/simplemath.h:4]: Consider not using Chinese Pinyin when naming a variable.

File: src/platform.h.in [hunk 8]
```diff
@@ -15,6 +15,7 @@
 #ifndef NCNN_PLATFORM_H
 #define NCNN_PLATFORM_H
 
+#cmakedefine01 NCNN_SIMPLEMATH
```
Comment (zchrissirhcz) [inline src/platform.h.in:18]: Consider moving `NCNN_SIMPLEPATH` to the bottom, just like the order when `src/CMakeLists.txt`.

**Comment by nihui** [comment]
pr closed as a much more superior simplemath implementation covered in https://github.com/Tencent/ncnn/pull/4905
thanks anyway !

**Pull request closed without merging**